src/cpp/command.cpp

```diff
@@ -119,6 +119,44 @@ void Command::add_field(std::string field, bool is_key)
     }
 }
 
+//  Replace a field in a command
+void Command::set_field_at(std::string new_field, size_t pos, bool is_key)
+{
+    size_t field_size = new_field.size();
+    char* f = NULL;
+    try {
+        f = (char*)new unsigned char[field_size + 1];
+    }
+    catch (std::bad_alloc& e) {
+        throw SRBadAllocException("field");
```

Comment:
Make this a little more descriptive.

Reply:
This is our standard for memory allocation failures.